include/pmacc/kernel/atomic.hpp

```diff
@@ -235,7 +235,8 @@ namespace pmacc
         DINLINE void atomicAllExch(const T_Acc& acc, T_Type* ptr, const T_Type value, const T_Hierarchy& hierarchy)
         {
             const auto mask = alpaka::warp::activemask(acc);
-            const auto leader = alpaka::ffs(acc, static_cast<std::make_signed_t<decltype(mask)>>(mask)) - 1;
+            [[maybe_unused]] const auto leader
+                = alpaka::ffs(acc, static_cast<std::make_signed_t<decltype(mask)>>(mask)) - 1;
```

Comment:
could you please use `alpaka::ignore_unused(leader);` to hide this warning. This will work for cpp14 too and not for cpp17+ only.

Reply:
Is it necessary to have this commit in this PR? This fix is mixing a feature with a warning fix. I would prefer to have a separate PR to be able to add correct lables.

Reply:
Done in #3803.